### PR TITLE
Adds support for DecimalNumberValueSerializer

### LINKED ISSUE #43: Amount is sent as number, consider string instead
I wanted to find another page which outlines in more depth the issues, but I could only find the ones below.

A quick summary is some JSON parsers will deserialize numbers as floats which obviously is problematic for Money. Other issues with eventually being deserialized as a float are limitations on size and problems with adding/subtracting. If you send it as a String it will be deserialized as a String. 

Various browsers use IEEE-754 to represent their numbers by default which would cause issues.

http://stackoverflow.com/questions/30249406/what-is-the-standard-for-formatting-currency-values-in-json

https://github.com/nlohmann/json/issues/98

https://github.com/shopspring/decimal/issues/21

### PR DESCRIPTION
**Pull request by whiskeysierra**
Fixes #43, ~~alternative to~~ based on #48

### CODE REVIEW

File: src/test/java/org/zalando/jackson/datatype/money/MonetaryAmountSerializerTest.java [hunk 1]
```diff
@@ -127,4 +130,25 @@ public void shouldSerializeWithCustomName() throws IOException {
         assertThat(actual, is(expected));
     }
 
-}
\ No newline at end of file
+    @Test
+    public void shouldSerializeAmountAsStringValue() throws JsonProcessingException {
```
Comment (AlexanderYastrebov) [inline src/test/java/org/zalando/jackson/datatype/money/MonetaryAmountSerializerTest.java]: asQuotedDecimal then :)

File: .java-version [hunk 2]
```diff
@@ -1 +0,0 @@
-1.7
```
Comment (AlexanderYastrebov) [inline .java-version]: Or drop file completely?

**Comment by AlexanderYastrebov** [comment]
Don't you want to make deserializer configurable as well?

File: .java-version (hunk 2 above)
Comment (whiskeysierra) [inline .java-version]: I did.

**Comment by whiskeysierra** [comment]
> Don't you want to make deserializer configurable as well?

I thought about it. Given the two built-in implementations for deserializing we *only* support decimal and quoted decimal. For serializing we support both, out of the box at the time already.

Now that we allow to pass in a custom serializer for NumberValue it kind of makes sense but I'd wait until someone requests it.

**Comment by AlexanderYastrebov** [comment]
:+1:

**Comment by Psynbiotik** [comment]
Looks good

**Pull request merged**